kiwi/builder/disk.py

```diff
@@ -175,33 +173,15 @@ def create(self):
         Image types which triggers this builder are:
 
         * image="oem"
-        * image="vmx"
 
         :return: result
 
         :rtype: instance of :class:`Result`
         """
-        disk = DiskBuilder(
-            self.xml_state, self.target_dir, self.root_dir, self.custom_args
-        )
-        result = disk.create_disk()
-
-        # cleanup disk resources taken prior to next steps
-        del disk
```

Comment:
This deletion of the disk instance needs to stay or some resource free must be issued. The reason is that when creating the install media or the disk format we should prevent open loop/mount setup into the system image at call time of these operations. There is a lot potential for strange effects if a loop is done over a loop or a squashfs filesystem is created over an active mounted filesystem. I know you might not have seen bad effects in your tests but this cleanup is there for a reason of past issues we had in that regard. I suggest to keep it

Reply:
Thanks for the comments, I realized there are a couple of issues with this refactor. Indeed, DiskBuiler creates some runtime structures that are never cleared (`LoopDevice` and `Disk`) unless the whole instance is cleared. I'll do another round of this refactor, however I am afraid this leads to bigger refactor, not in terms the amount of code, but in terms of sensitive changes.